internal/helm/client/client.go

```diff
@@ -106,11 +106,9 @@ func NewOwnerRefInjectingClient(base kube.Client, restMapper meta.RESTMapper,
 	cr *unstructured.Unstructured) (kube.Interface, error) {
 
 	if cr != nil {
-		if cr.GetObjectKind() != nil {
-			if cr.GetObjectKind().GroupVersionKind().Empty() || cr.GetName() == "" || cr.GetUID() == "" {
-				var err = errors.New("owner resource is invalid")
-				return nil, err
-			}
+		if cr.GetObjectKind().GroupVersionKind().Empty() || cr.GetName() == "" || cr.GetUID() == "" {
```

Comment:
Just curious about this change. The nesting was intentional, because `GetObjectKind()` returns `*unstructured.Unstructured` (https://github.com/kubernetes/apimachinery/blob/97e5df2d0258ac077093867fabf508c1fc31f53b/pkg/apis/meta/v1/unstructured/unstructured.go#L52) and if that is nil, then this would cause nil pointer error. If something changed in 1.24 version of apimachinery to handle this, then its good.

Reply:
I see. `schema.ObjectKind` is supposed to be an interface, and an unstructured obj just implements it. But if we are checking that `cr` is not nil somewhere then its good.